dnf-behave-tests/dnf/steps/repo.py

```diff
@@ -372,3 +378,22 @@ def given_system(context, system):
 def given_no_osrelease(context):
     behave.use_fixture(osrelease_fixture, context)
     context.scenario.osrelease.delete()
+
+
+@behave.step("I rewrite solvfile version of \"{path}\" with \"{version}\"")
+def rewrite_solvfile_version(context, path, version):
+    assert len(version) == 3, "Version is expected to be 3 characters long"
+    path = path.format(context=context)
+    ensure_file_exists(path)
+    fp = open(path, "r+b")
+    # Seek 40 bytes from start of file to the beginning userdata section
+    fp.seek(40, 0)
+    userdata_dnf_magic = fp.read(4).decode('utf-8')
+    assert userdata_dnf_magic == "dnfS"
+
+    userdata_dnf_version = fp.read(4).decode('utf-8')
+    assert userdata_dnf_version == "01.0"
```

Comment:
This now relies on the libsolv solvfile format and dnf userdata section format, it will start failing when there is a change.

Unfortunately I cannot think of any better solution.